browser/brave_vpn/dns/brave_vpn_dns_observer_factory.h

```diff
@@ -0,0 +1,43 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_BRAVE_VPN_DNS_BRAVE_VPN_DNS_OBSERVER_FACTORY_H_
+#define BRAVE_BROWSER_BRAVE_VPN_DNS_BRAVE_VPN_DNS_OBSERVER_FACTORY_H_
+
+#include "base/memory/singleton.h"
+#include "components/keyed_service/content/browser_context_keyed_service_factory.h"
+
+namespace content {
+class BrowserContext;
+}  // namespace content
+
+namespace brave_vpn {
+
+class BraveVpnDnsObserverService;
+
+class BraveVpnDnsObserverFactory : public BrowserContextKeyedServiceFactory {
```

Comment:
this class is named incorrectly based on the service name. It should be BraveVpnDnsObserverServiceFactory

Reply:
also why does this need to be a separate keyed service and what does it do?